acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/hashicorpVault/HashicorpVaultDocker.java

```diff
@@ -0,0 +1,222 @@
+import com.github.dockerjava.api.exception.NotFoundException;
+import com.github.dockerjava.api.exception.NotModifiedException;
+import com.github.dockerjava.api.model.Capability;
+import com.github.dockerjava.api.model.ExposedPort;
+import com.github.dockerjava.api.model.HostConfig;
+import com.github.dockerjava.api.model.PortBinding;
+import com.github.dockerjava.api.model.Ports;
+import com.github.dockerjava.api.model.Ports.Binding;
+import com.github.dockerjava.core.command.ExecStartResultCallback;
+import com.github.dockerjava.core.command.PullImageResultCallback;
+import com.github.dockerjava.core.command.WaitContainerResultCallback;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class HashicorpVaultDocker {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String HASHICORP_VAULT_IMAGE = "vault:latest";
+  private static final int DEFAULT_HTTP_PORT = 8200;
+  public static final String vaultToken = "token";
+  private static final String[] CREATE_ETHSIGNER_SIGNING_KEY_SECRET = {
+    "vault",
+    "kv",
+    "put",
+    "secret/ethsignerSigningKey",
+    "value=8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63"
+  };
+
+  private final DockerClient docker;
+  private final String vaultContainerId;
+
+  private int port;
+
+  public HashicorpVaultDocker(final DockerClient docker) {
+    this.docker = docker;
+    pullVaultImage();
+    this.vaultContainerId = createVaultContainer();
+  }
+
+  public void start() {
+    LOG.info("Starting Hashicorp Vault Docker container: {}", vaultContainerId);
+    docker.startContainerCmd(vaultContainerId).exec();
+
+    LOG.info("Querying for the Docker dynamically allocated vault port number");
+    final InspectContainerResponse containerResponse =
+        docker.inspectContainerCmd(vaultContainerId).exec();
+    final Ports ports = containerResponse.getNetworkSettings().getPorts();
+    port = httpRpcPort(ports);
+    LOG.info("Http port for Hashicorp Vault: {}", port);
+
+    try {
+      TimeUnit.SECONDS.sleep(10); // wait until the docker is container is all up and running.
+    } catch (InterruptedException e) {
+      LOG.error("Interruption while waiting for Hashicorp Vault to be up");
+    }
+
+    // After starting the docker container with the vault we need to create the secret that contains
+    // the
+    // the private key. That is done in the awaitStartupCompletion, because we need to wait until a
+    // call to
+    // the vault succeeds to know that it is up. That call now creates the secret.
+  }
+
+  private ExecCreateCmdResponse getExecCreateCmdResponse(String[] commandWithArguments) {
+    return docker
+        .execCreateCmd(vaultContainerId)
+        .withAttachStdout(true)
+        .withAttachStderr(true)
+        .withCmd(commandWithArguments)
+        .exec();
+  }
+
+  private boolean runCommandInVaultContainer(
```

Comment:
nit; `final` modifier can be added to these input & method variables

Reply:
Done